stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/search/SearchResultListAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package io.getstream.chat.android.ui.search
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.ui.databinding.StreamItemMentionListBinding
+import io.getstream.chat.android.ui.messagepreview.MessagePreviewView
+import io.getstream.chat.android.ui.search.SearchResultListAdapter.MessagePreviewViewHolder
+import io.getstream.chat.android.ui.search.SearchResultListView.SearchResultSelectedListener
+
+public class SearchResultListAdapter : ListAdapter<Message, MessagePreviewViewHolder>(MessageDiffCallback) {
+
+    private var searchResultSelectedListener: SearchResultSelectedListener? = null
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MessagePreviewViewHolder {
+        return StreamItemMentionListBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let { MessagePreviewViewHolder(it.root) }
+    }
+
+    override fun onBindViewHolder(holder: MessagePreviewViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    public fun setSearchResultSelectedListener(searchResultSelectedListener: SearchResultSelectedListener?) {
```

Comment:
Is `searchResultSelectedListener` nullable on purpose?

Reply:
It is! It's null by default anyway, and this way you can unset it by passing in null, if you want to.